### PR TITLE
(case 22201) Fix Interface not loading some baked models correctly

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.manuscript.com/f/cases/22201/Baking-some-models-in-Oven-causes-their-materials-to-get-mixed-up

This PR fixes a logic bug allowing the new baked models to load their materials correctly when mesh part material definitions are in a nontrivial order.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18751/HighFidelity-Beta-PR15399-325ea50.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18751/HighFidelity-Quest-Beta-PR15399-325ea50.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18751/HighFidelity-Beta-PR15399-325ea50.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18751/HighFidelity-Beta-Interface-PR15399-325ea50.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18751/HighFidelity-Beta-PR15399-325ea50.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18751/HighFidelity-Beta-Interface-PR15399-325ea50.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18755/HighFidelity-Beta-PR15399-cf685d5.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18755/HighFidelity-Quest-Beta-PR15399-cf685d5.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18755/HighFidelity-Beta-PR15399-cf685d5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18755/HighFidelity-Beta-Interface-PR15399-cf685d5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18755/HighFidelity-Beta-PR15399-cf685d5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18755/HighFidelity-Beta-Interface-PR15399-cf685d5.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18765/HighFidelity-Beta-PR15399-7f42eca.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18765/HighFidelity-Beta-Interface-PR15399-7f42eca.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18765/HighFidelity-Beta-PR15399-7f42eca.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18765/HighFidelity-Quest-Beta-PR15399-7f42eca.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18765/HighFidelity-Beta-PR15399-7f42eca.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18765/HighFidelity-Beta-Interface-PR15399-7f42eca.exe

File: libraries/fbx/src/FBXSerializer_Mesh.cpp [hunk 1]
```diff
@@ -492,7 +492,7 @@ ExtractedMesh FBXSerializer::extractMesh(const FBXNode& object, unsigned int& me
                     // Figure out what material this part is
                     if (dracoMeshNodeVersion >= 2) {
                         // Define the materialID now
-                        if (dracoMaterialList.size() - 1 <= materialID) {
+                        if (materialID <= dracoMaterialList.size() - 1) {
```
Comment (Atlante45) [inline libraries/fbx/src/FBXSerializer_Mesh.cpp]: Not a blocker, but `materialID < dracoMaterialList.size()` feels like a more common way to write this.
Should we also check that it's not negative?

Comment (sabrina-shanman) [reply libraries/fbx/src/FBXSerializer_Mesh.cpp]: Yes for the first.
materialID is unsigned.

**Pull request merged**